qcoro/qml/qcoroqmltask.cpp

```diff
@@ -73,4 +73,24 @@ void QmlTask::then(QJSValue func) {
     });
 }
 
+QmlTaskListener *QmlTask::await()
+{
+    auto *listener = new QmlTaskListener();
+    d->task->then([listener](auto &&value) {
+        listener->setValue(std::move(value));
+    });
```

Comment:
Since the `listener` object life time will be managed by the QML engine, there might be a case when the `listener` is destroyed by the QML engine before the Task finishes (e.g. due to unloading a QML component) which would lead to a crash when the task finishes.

I think you need to capture the listener into a QPointer here and check for its validity before calling `setValue`.